internal/storage/local/local.go

```diff
@@ -0,0 +1,137 @@
+package local
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/offen/docker-volume-backup/internal/storage"
+	utilites "github.com/offen/docker-volume-backup/internal/utilities"
+)
+
+type localStorage struct {
+	*storage.StorageBackend
+	latestSymlink string
+}
+
+// NewStorageBackend creates and initializes a new local storage backend.
+func NewStorageBackend(archivePath string, latestSymlink string, logFunc storage.LogFuncDef) storage.Backend {
+
+	strgBackend := &storage.StorageBackend{
+		Backend:         &localStorage{},
+		Name:            "Local",
+		DestinationPath: archivePath,
+		Log:             logFunc,
+	}
+	localBackend := &localStorage{
+		StorageBackend: strgBackend,
```

Comment:
This seems a little twisted: what's the reason we need the reference back to the `strgBackend` object, creating a circular structure?

Reply:
Initializing structs that embed other structs is a little non-obvious in Go. If I understand what you're trying to achieve here correctly, you could also do it like this:

```go
type localStorage struct {
	storage.StorageBackend
	latestSymlink string
}

// NewStorageBackend creates and initializes a new local storage backend.
func NewStorageBackend(archivePath string, latestSymlink string, logFunc storage.LogFuncDef) storage.Backend {
	return &localStorage{
		StorageBackend: storage.StorageBackend{
			Name:            "Local",
			DestinationPath: archivePath,
			Log:             logFunc,
		},
		latestSymlink: latestSymlink,
	}
}
```

Even when you declare `Name` et al on `StorageBackend` they will propagate to the top level as the type is embedded.

Reply:
Thanks for the example, that indeed looks way better and simpler.